l3kernel/l3doc.dtx

```diff
@@ -2751,11 +2751,27 @@ and all files in that bundle must be distributed together.
     \int_compare:nTF { \seq_count:N \g_@@_variants_seq == 1 }
       { \seq_use:Nn \g_@@_variants_seq { } }
       {
+       \hbox_set:Nn\l_tmpa_box
+         { \seq_use:Nn \g__@@_variants_seq { \textrm|\nolinebreak[2]} }
         \textrm(
-          \seq_use:Nn \g_@@_variants_seq { \textrm| }
-        \textrm)
+%    \end{macrocode}
+%
+%   Set long variant lists in a parbox, short lists set natural length.
+%    \begin{macrocode}
+         \dim_compare:nNnTF {\wd\l_tmpa_box}>{.4\columnwidth}
+          {
+          \parbox[t]{.4\columnwidth}
+             {\raggedright
+               \hbox_unpack_drop:N\l_tmpa_box
+               \textrm)
+               \bool_if:NT #2 { \@@_typeset_TF: }
+             }
+          }
+          { \hbox_unpack_drop:N\l_tmpa_box
+             \textrm)
+             \bool_if:NT #2 { \@@_typeset_TF: }
+          }
```

Comment:
Not sure if you like this indent and spacing
```suggestion
        \dim_compare:nNnTF { \box_wd:N \l_tmpa_box } > { .4\columnwidth }
          {
            \parbox[t]{.4\columnwidth}
              {
                \raggedright
                \hbox_unpack_drop:N \l_tmpa_box
                \textrm)
                \bool_if:NT #2 { \@@_typeset_TF: }
              }
          }
          {
            \hbox_unpack_drop:N \l_tmpa_box
            \textrm)
            \bool_if:NT #2 { \@@_typeset_TF: }
          }
      }
```

Reply:
I think this one's got an extra } at the end, I'll apply locally

Reply:
Ya you're right. I pasted too much 😅 (from a locally modified copy of the whole `\int_compare:nTF { \seq_count:N \g_@@_variants_seq == 1 }` block).

Reply:
no problem, this got done as 101dbb4514a9da07b447eec1610afb721e80cfc9